smarts/core/sumo_road_network.py

```diff
@@ -85,18 +88,81 @@ def __init__(self, graph):
         self._log = logging.getLogger(self.__class__.__name__)
         self._graph = graph
 
+    @staticmethod
+    def _check_net_origin(bbox):
+        assert len(bbox) == 4
+        return bbox[0] <= 0.0 and bbox[1] <= 0.0 and bbox[2] >= 0.0 and bbox[3] >= 0.0
+
+    @classmethod
+    @lru_cache(maxsize=1)
+    def _shift_coordinates(cls, net_file):
+        logger = logging.getLogger(cls.__name__)
+        logger.info("normalizing net coordinates...")
+        with NamedTemporaryFile() as tf:
+            ## Translate the map's origin to remove huge (imprecise) offsets.
+            ## See https://sumo.dlr.de/docs/netconvert.html#usage_description
+            ## for netconvert options description.
+            try:
+                check_call(
+                    [
+                        "netconvert",
+                        "--offset.disable-normalization=FALSE",
+                        "-s",
+                        net_file,
+                        "-o",
+                        tf.name,
+                    ]
+                )
```

Comment:
Any idea on the cost of this call? Since this is cashed it should be low assuming we are only working with a single map but that is not always the case. 

The worst case that should happen here is if we have multiple maps, as with ULTRA, and the cache has to be reset every time.

Reply:
Although I don't know how the `netconvert` tool scales with network complexity, I ran several tests shifting the networks attached to the original issue.  For these, the average appears to be about:
```
real	0m0.072s
user	0m0.208s
sys	0m0.183s
```

If this is too much, we could increment the number that we cache to handle the ULTRA case too, like `@lru_cache(4)`, although if the networks consume a lot of memory it might not be desirable to keep more than one of them around.